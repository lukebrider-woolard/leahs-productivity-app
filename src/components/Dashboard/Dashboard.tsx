import PageLayout from '../PageLayout/PageLayout';

export default function Dashboard() {

  function render() {
    return (
      <></>
    )
  }

  return (
    <PageLayout pageTitle='Dashboard' child={render()} />
  )
}
