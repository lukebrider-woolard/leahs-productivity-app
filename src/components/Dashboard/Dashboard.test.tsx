import Dashboard from './Dashboard';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

// Mock localDataUtils functions as localStorage mock doesn't work well.

beforeEach(() => {
  jest.clearAllMocks();
});

describe('Dashboard UI Tests', () => {
  test('dashboard page renders default view when data not initialised', () => {
    const dashboardInitTree = renderer.create(<Dashboard />).toJSON();

    expect(dashboardInitTree).toMatchSnapshot();
  });

  test('dashboard page renders new buttons after data initialised', async () => {
    render(<Dashboard />);
    const initButton = screen.getByRole('button', {
      name: /initialise app data/i,
    });

    userEvent.click(initButton);

    await waitFor(() => {
      expect(screen.getByText(/reset local storage/i)).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(
        screen.getByText(/write stored data to console/i)
      ).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(screen.queryByText(/initialise app data/i)).toBeNull();
    });
  });

  // test init button calls right function with expected

  // test write data button calls right function with expected

  // test clear data button calls right function with expected
});
