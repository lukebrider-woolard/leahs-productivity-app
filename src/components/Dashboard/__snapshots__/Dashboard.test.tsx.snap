// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard UI Tests dashboard page renders default view when data not initialised 1`] = `
<div
  className="MuiBox-root css-8875ym"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 css-1bt2q07-MuiTypography-root"
  >
    Dashboard
  </h4>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-qhkpqj-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-s5m0mu"
    >
      <div
        className="MuiStack-root css-10y7f4i-MuiStack-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-rs4doj-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Initialise App Data
        </button>
      </div>
    </div>
  </div>
</div>
`;
